scripts/coverage_report.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+# Check that the test directory and report path arguments are provided
+if [ $# -lt 2 ]
+then
+	echo "Usage: $0 <testdir> <outdir>"
+	echo "All tests in <testdir> and its subdirectories will be run to measure coverage."
+	echo "The resulting coverage report will be stored in <outdir>."
+	exit 1
+fi
+
+# User prompts will be skipped if '--batch' is given as the third argument
+SKIP_PROMPTS=0
+if [ $# -eq 3 -a "$3" == "--batch" ]
+then
+	SKIP_PROMPTS=1
+fi
+
+# Set the directory containing the tests to run (includes subdirectories)
+TEST_DIR=$1
+
+# Set the directory to which the report will be saved
+COVERAGE_DIR=$2
+
+# This needs to run in libra
+LIBRA_DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && pwd)"
+if [ $(pwd) != $LIBRA_DIR  ]
+then
+	echo "This needs to run from libra/, not in $(pwd)"
+	exit 1
+fi
+
+set -e
+
+# Check that grcov is installed
+if ! [ -x "$(command -v grcov)" ]; then
+	echo "Error: grcov is not installed." >&2
+	read -p "Install grcov? [yY/*] " -n 1 -r
```

Comment:
Could we protect this with `SKIP_PROMPTS` (in general, in batch mode error message + `exit 1` is the better remediation to an improper installation).